### ISSUE TITLE
[Doc] Create CONTRIBUTING.md

### ISSUE THREAD

**Issue by mkubdev** [TheAlgorithms/Solidity#13]
Create `Contributing guidelines` file for the Solidity repos.

* We should make clear what coding style we'll use to format commit of the repos. 
* We also need to discuss how to define the folders structure.

💡: [Python](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md)'s one is awesome !

**Comment by akhil484** [comment]
Hey, I would like to contribute to this issue and can help in creating contributing guidelines. Can we discuss it a bit more?

**Comment by akhil484** [comment]
We can take Python's contributing.md that you mentioned above as a template and can discuss all areas we should include. How can we discuss it at length?

**Comment by akhil484** [comment]
Yes, I have sent you request on Discord. Let's discuss the coding style and what instructions should we provide to contributors.

**Comment by mkubdev** [comment]
Ok accepted.
I think we can organize smart contracts by networks :

```md
├── Ethereum
│   ├── Maths
│   ├── Common
│   ├── ...
├── Polygon
│   ├── Maths
│   ├── Common
│   │   ├── governance
│   │   ├── tokens
│   │   ├── misc
│   ├── Staking
│   ├── ...
├── ...
└── .gitignore
```

I wait for contributors answer. 🔈

**Comment by shuklaritvik06** [comment]
> Ok accepted. I think we can organize smart contracts by networks (Networks have somehow different business logic..) :
> 
> ```lisp
> ├── Ethereum
> │   ├── Maths
> │   ├── Common
> │   ├── ...
> ├── Polygon
> │   ├── Maths
> │   ├── Common
> │   │   ├── governance
> │   │   ├── tokens
> │   │   ├── misc
> │   ├── Staking
> │   ├── ...
> ├── ...
> └── .gitignore
> ```
> 
> I'm waiting for the opinion of contributors. speaker

I agree!

**Comment by akhil484** [comment]
I think organising smart contract by networks looks better.

**Comment by KukretiShubham** [comment]
> Or we organize contracts by Solidity Concepts like:
> 
> ```lisp
> ├── Maths
> │   ├── ...
> ├── Security
> │   ├── ...
> ├── DAO
> │   ├── ...
> ├── Governance
> ├── Staking
> ```
> 
> I don't know if it's a better approach instead of storing concepts by network since they can be implemented in every EVM networks.

I agree solidity is EVM compatible language. So code execution will work exactly same on all networks. I like @mkubdev 's suggestion

**Comment by mkubdev** [comment]
> So code execution will work exactly the same on all networks

That's what I thought! Thanks.

**Comment by mkubdev** [comment]
I close, solved by #17
